pkg/image/api/validation/validation.go

```diff
@@ -43,6 +44,17 @@ func ValidateImageStream(stream *api.ImageStream) fielderrors.ValidationErrorLis
 
 	result = append(result, validation.ValidateObjectMeta(&stream.ObjectMeta, true, MinimalNameValidation).Prefix("metadata")...)
 
+	// Ensure we can generate a valid docker image repository from namespace/name
```

Comment:
This is technically only relevant if the stream is using our internal registry. If spec.dockerImageRepository is set, it's ok to have a 1-char ns or name, since they won't be used in the push/pull spec. Not sure if it's worth making that distinction in the code, however.